structure/sqrtdecompositionsimple/sqrtdecompositionsimple_test.go

```diff
@@ -0,0 +1,79 @@
+package sqrtdecompositionsimple
```

Comment:
Use the black box testing method for this structure, this is a powerful structure but very fiddly so there may be things that you may have overlooked until you test it as a consumer.

```suggestion
package sqrtdecompositionsimple_test
```
or add `_test` to package name when you change the package name.